msal/src/main/java/com/microsoft/identity/client/AuthenticationActivity.java

```diff
@@ -65,6 +65,8 @@ protected void onCreate(final Bundle savedInstanceState) {
         if (savedInstanceState != null) {
             Logger.verbose(TAG, null, "AuthenticationActivity is re-created after killed by the os.");
             mRestarted = true;
+            mTelemetryRequestId = savedInstanceState.getString(Constants.TELEMETRY_REQUEST_ID);
+            mUiEventBuilder = new UiEvent.Builder();
```

Comment:
seems like you can move the mUiEventBuilder before the check for saveInstanceState != null